pandas/io/s3.py

```diff
@@ -25,7 +26,15 @@ def get_file_and_filesystem(
     if mode is None:
         mode = "rb"
 
-    fs = s3fs.S3FileSystem(anon=False)
+    # Support customised S3 servers endpoint URL via environment variable
+    # The S3_ENDPOINT should be the complete URL to S3 service following
+    # the format: http(s)://{host}:{port}. If S3_ENDPOINT is undefined, it will
+    # fallback to use the default AWS S3 endpoint as determined by boto3.
+    s3_endpoint = os.environ.get("S3_ENDPOINT")
+    client_kwargs: Optional[Dict[str, str]] = {
```

Comment:
wouldn't this be better as a feature directly to s3fs?

Reply:
Well, here comes an awkward situation: From the point view of s3fs and boto3, they already support customised endpoint by accepting parameters `endpoint_url` via `client_kwargs`, so there is no point to add environment variable support for this parameter there, as discussed on this [boto3 issue](https://github.com/boto/boto3/issues/1375#issuecomment-349455862). 

The problem with Pandas now is that it doesn't provide a way to pass this parameter over to s3fs. I think adding support using this environment variable will benefit many people using their own s3 servers.

Reply:
Why wouldn't using the environment variable as the default endpoint_url go in s3fs, to benefit even more people?